doc/Ontology.md

```diff
@@ -0,0 +1,208 @@
+# ACM Knowledge Graph Ontology
+
+## Overview
+
+For details on Knowledge Graphs you can reference other resources.  Perhaps
+start with the [README.md](../README.md) in this repository to begin 
+learning more.
+
+To get started with Knowledge Graphs in ACM, we want our nodes to consist
+of clearly defined entities.  Where possible we want to always re-use the
+existing entities that have been defined instead of creating new ones.
+Validation is being put in place to help prevent too many entities from
+being created.  Too many entity types will make knowledge graph contributions 
+and maintenance more difficult.
+
+## Entities
+
+The list of entities are validated from the [Glossary](../glossary.txt) file
+and consist of the common resources shown below.
+
+![](./ontology.png)
+
+See the examples in the [input](../input) folder for examples on how this
+ontology is being used.
+
+## Entity Descriptions
+
+Entities are the nodes in the knowledge graph.  The following is a list of the
+main entities that are being used in the samples.  While more entities can
+be added, they must not duplicate the ones that already exist.
+
+Entity | Description | Samples | Challenges 
+---    | ---         | ---     | ---
+Users | Users of the ACM APIs. This would also include groups and different User types, such as usage by other Operators or teams. | Customer, Developer, QE, Support | 
+API | A public facing API for the system. Commonly this is the ACM Custom Resource Definitions.  This can also be extended to other APIs such as OpenShift APIs or Non K8S APIs like REST, Ansible and cloud platforms. | ManagedCluster, Policy, Placement, Ansible Playbook | This can seem to overlap with Custom Resource and Intermediate Resource which are also API based. 
+Intermediate Resource | An instance of a resource that was not directly created by a User. | Pods, Deployments, Secrets, ConfigurationPolicy | 
+Processor | A running process that manipulates resources | Any pod or operator or even an external rest server/engine. | 
+Gateway | A route that allows external access to the system | An OpenShift route, A port on a node, a service that is accessible | This is currently intended to be an OpenShift route resource.  Extend as necessary.
+Repository | A storage mechanism for code or other resources such as policies or applications. | github.com/organization/repository-name is a full path to a particular project or repository |
+
+## Attributes of Entities
+
+Each entity (node) in the graph will have multiple attributes to help clarify the details
+of the node.  This table lists the main attributes.  You should use as many
+as possible to help make sure your entities are uniquely defined if they
```

Comment:
To avoid confusion should we say:
`You should use as many` 
or
`You should use all`

Reply:
Updated this too!